pkgs/by-name/gl/globalplatform/package.nix

```diff
@@ -0,0 +1,81 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pcsclite,
+  PCSC,
+  pkg-config,
+  cmake,
+  zlib,
+  pandoc,
+  doxygen,
+  graphviz,
+  openssl,
+  cmocka,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "globalplatform";
+  version = "2.4.0-unreleased";
+
+  src = fetchFromGitHub {
+    owner = "kaoh";
+    repo = "globalplatform";
+    rev = "0f970751c5d9e8a7030f897ca2d1b86d0eeba4c2";
+    sha256 = "sha256-H/muc/gY5glXPWKj75fHi6+1DAP91YGAUefdQkX9nfk=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    pandoc
+    doxygen
+    graphviz
+  ];
+
+  buildInputs =
+    [
+      zlib
+      openssl
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      pcsclite
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      PCSC
+    ];
+
+  cmakeFlags = [
+    "-DTESTING=ON"
+  ];
+
+  doCheck = true;
+
+  nativeCheckInputs = [
+    cmocka
+  ];
+
+  preCheck = ''
+    cp "$src/gpshell/helloworld.cap" globalplatform/src
+    cp "$src/globalplatform/src/rsa_pub_key_test.pem" globalplatform/src
+  '';
+
+  # libglobalplatform.so uses dlopen() to load specified connection plugins at runtime.
+  # Currently, libgppcscconnectionplugin.so is the only plugin included.
+  # The user has to specify custom plugin locations by setting LD_LIBRARY_PATH.
+
+  postFixup = ''
+    patchelf $out/lib/libglobalplatform.so --add-rpath "$out/lib"
```

Comment:
Darwin builds seems to be failing on this. I'm not familiar enough with darwin to suggest a solution. Is it common for darwin to use patchelf?

Reply:
I have added a conditional call to `install_name_tool`, which seems to be the equivalent for Darwin. I'll see what the builder says to that.

Reply:
Well, seem like the darwin runners are happy now :)